invenio/modules/search/enhancers/cache_results.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Query results cacher."""
+
+from invenio.modules.search.cache import get_results_cache, set_results_cache
+
+
+class CacheOp(object):
+
+    """Store results in cache."""
```

Comment:
For now it's using same functions `(get|set)_results_cache` like in 2.0 (https://github.com/inveniosoftware/invenio/pull/3068#discussion-diff-29278110R22).